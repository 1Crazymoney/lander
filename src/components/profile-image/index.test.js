import React from 'react';
import ProfileImage from './index';
import renderer from 'react-test-renderer';

import wrapWithIntl from '../../utils/test-helper';

it('not in edit mode and image is empty. should render empty image svg ', () => {
  const props = {};

  const component = wrapWithIntl(<ProfileImage {...props} />);

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('in edit mode and image is empty. should render empty image svg and edit btn', () => {
  const props = {
    editMode: true
  };

  const component = wrapWithIntl(<ProfileImage {...props} />);

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('not in edit mode and image is not empty. should render image ', () => {
  const props = {
    image: [{
      contentUrl: 'https://foo.bar/baz.jpg'
    }]
  };

  const component = wrapWithIntl(<ProfileImage {...props} />);

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('in edit mode and image is not empty. should render image and edit btn', () => {
  const props = {
    editMode: true,
    image: [{
      contentUrl: 'https://foo.bar/baz.jpg'
    }]
  };

  const component = wrapWithIntl(<ProfileImage {...props} />);

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});