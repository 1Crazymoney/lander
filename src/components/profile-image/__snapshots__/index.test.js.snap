// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`in edit mode and image is empty. should render empty image svg and edit btn 1`] = `
<div
  className="profile-image"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-image fa-w-16"
    data-icon="image"
    data-prefix="far"
    focusable="false"
    role="img"
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm-6 336H54a6 6 0 0 1-6-6V118a6 6 0 0 1 6-6h404a6 6 0 0 1 6 6v276a6 6 0 0 1-6 6zM128 152c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zM96 352h320v-80l-87.515-87.515c-4.686-4.686-12.284-4.686-16.971 0L192 304l-39.515-39.515c-4.686-4.686-12.284-4.686-16.971 0L96 304v48z"
      fill="currentColor"
    />
  </svg>
   
  <div
    className="edit-btn"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-pen fa-w-16"
      data-icon="pen"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`in edit mode and image is not empty. should render image and edit btn 1`] = `
<div
  className="profile-image"
  style={
    Object {
      "backgroundImage": "url('https://foo.bar/baz.jpg')",
    }
  }
>
  <div
    className="edit-btn"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-pen fa-w-16"
      data-icon="pen"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`not in edit mode and image is empty. should render empty image svg  1`] = `
<div
  className="profile-image"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-image fa-w-16"
    data-icon="image"
    data-prefix="far"
    focusable="false"
    role="img"
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm-6 336H54a6 6 0 0 1-6-6V118a6 6 0 0 1 6-6h404a6 6 0 0 1 6 6v276a6 6 0 0 1-6 6zM128 152c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zM96 352h320v-80l-87.515-87.515c-4.686-4.686-12.284-4.686-16.971 0L192 304l-39.515-39.515c-4.686-4.686-12.284-4.686-16.971 0L96 304v48z"
      fill="currentColor"
    />
  </svg>
   
</div>
`;

exports[`not in edit mode and image is not empty. should render image  1`] = `
<div
  className="profile-image"
  style={
    Object {
      "backgroundImage": "url('https://foo.bar/baz.jpg')",
    }
  }
/>
`;
